components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/AuthenticatorEndpointConfiguration.java

```diff
@@ -0,0 +1,141 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model;
+
+import org.wso2.carbon.identity.action.management.model.Authentication;
+import org.wso2.carbon.identity.action.management.model.Authentication.Property;
+import org.wso2.carbon.identity.action.management.model.EndpointConfig;
+
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+
+/**
+ * The authenticator endpoint configuration model.
+ */
+public class AuthenticatorEndpointConfiguration {
+
+    private String uri;
+    private String authenticationType;
+    private HashMap<String, String> authenticationProperties;
+
+    public AuthenticatorEndpointConfiguration() {
+    }
+
+    public AuthenticatorEndpointConfiguration(AuthenticatorEndpointConfigurationBuilder builder) {
+
+        this.uri = builder.uri;
+        this.authenticationProperties = builder.authenticationProperties;
+        this.authenticationType = builder.authenticationType;
+    }
+
+    public String getUri() {
+
+        return uri;
+    }
+
+    public HashMap<String, String> getAuthenticationProperties() {
+
+        return authenticationProperties;
+    }
+    
+    public String getAuthenticationType() {
+
+        return authenticationType;
+    }
+
+    /**
+     * AuthenticatorEndpointConfiguration builder.
+     */
+    public static class AuthenticatorEndpointConfigurationBuilder {
+
+        private String uri;
+        private String authenticationType;
+        private HashMap<String, String> authenticationProperties;
+
+        public AuthenticatorEndpointConfigurationBuilder() {
+        }
+
+        public AuthenticatorEndpointConfigurationBuilder uri(String uri) {
+
+            this.uri = uri;
+            return this;
+        }
+
+        public AuthenticatorEndpointConfigurationBuilder authenticationProperties(
+                HashMap<String, String> authentication) {
+
+            this.authenticationProperties = authentication;
+            return this;
+        }
+
+        public AuthenticatorEndpointConfigurationBuilder authenticationType(String authenticationType) {
+
+            this.authenticationType = authenticationType;
+            return this;
+        }
+        
+        public AuthenticatorEndpointConfiguration build() {
+
+            try {
+                EndpointConfig.EndpointConfigBuilder actionEndpointConfigBuilder =
+                        new EndpointConfig.EndpointConfigBuilder();
+                actionEndpointConfigBuilder.uri(this.uri);
+                actionEndpointConfigBuilder.authentication(buildAuthentication(
+                        this.authenticationType, this.authenticationProperties));
+                actionEndpointConfigBuilder.build();
+            } catch (Exception e) {
```

Comment:
Doesn't this builder throw a particular exception type ? If it's a runtime exception it's anyway thrown

Reply:
Addressed with commit 576fa6a82c23b1a93941ee4fb4edab63e8c85cef